sql/src/main/java/io/crate/expression/symbol/format/SymbolPrinter.java

```diff
@@ -182,9 +212,92 @@ private String anyOperatorName(String functionName) {
+                }
+            }
+            sb.append(")");
+            sb.append(COMMA).append(WS);
+            // second argument (keyword)
+            process(arguments.get(1), context);
+            sb.append(")");
+
+            // third argument (method)
+            sb.append(" USING ");
+            // need to print as identifier without quotes
+            sb.append(((BytesRef) ((Literal) arguments.get(2)).value()).utf8ToString());
+
+            // fourth argument (properties)
+            Map properties = (Map) ((Literal) arguments.get(3)).value();
+            if (properties.isEmpty()) {
+                return;
+            }
+            sb.append(" WITH (");
+            Iterator<Map.Entry> propertiesIterator = properties.entrySet().iterator();
+            while (propertiesIterator.hasNext()) {
+                Map.Entry entry = propertiesIterator.next();
+                sb.append(entry.getKey()).append(" = ");
+                Object value = entry.getValue();
+                if (value != null) {
+                    sb
+                        .append(WS)
+                        .append("'");
+                    if (value instanceof BytesRef) {
+                        sb.append(((BytesRef) value).utf8ToString());
+                    } else {
+                        sb.append(value);
+                    }
+                    sb.append("'");
+                }
+                if (propertiesIterator.hasNext()) {
+                    sb.append(COMMA).append(WS);
+                }
+            }
+            sb.append(")");
+        }
+
+        private void printSubscriptFunction(Function function, SymbolPrinterContext context) {
+            List<Symbol> arguments = function.arguments();
+            if (arguments.get(0) instanceof Reference &&
+                    arguments.get(0).valueType() instanceof ArrayType &&
+                    ((Reference) arguments.get(0)).ident().columnIdent().path().size() > 0) {
+                Reference firstArgument = (Reference) arguments.get(0);
+                context.builder.append(firstArgument.ident().columnIdent().name());
+                context.builder.append("[");
+                process(arguments.get(1), context);
+                context.builder.append("]");
+                context.builder.append("['");
+                context.builder.append(firstArgument.ident().columnIdent().path().get(0));
+                context.builder.append("']");
+            } else {
+                process(arguments.get(0), context);
+                context.builder.append("[");
+                process(arguments.get(1), context);
+                context.builder.append("]");
+            }
+        }
+
         @Override
         public Void visitReference(Reference symbol, SymbolPrinterContext context) {
-            if (context.isFullQualified()) {
+            if (context.isFullQualified() && !symbol.ident().tableIdent().equals(UnnestFunction.TABLE_IDENT)) {
```

Comment:
why do we have to handle unnest special but not other table functions?

Reply:
Hmm good point. This should generally apply to table functions.